httpcore/_async/http_proxy.py

```diff
@@ -39,15 +32,15 @@ class AsyncHTTPProxy(AsyncConnectionPool):
     def __init__(
         self,
         proxy_origin: Origin,
+        proxy_mode: str,
         proxy_headers: Headers = None,
-        proxy_mode: str = "DEFAULT",
         ssl_context: SSLContext = None,
         max_connections: int = None,
         max_keepalive: int = None,
         keepalive_expiry: float = None,
         http2: bool = False,
     ):
-        assert proxy_mode in ("DEFAULT", "FORWARD_ONLY", "TUNNEL_ONLY")
+        assert proxy_mode in ("FORWARD", "TUNNEL")
```

Comment:
I removed the `_ONLY` suffix here which does not match the values in HTTPX but more semantic sense.

Reply:
We might want to leave these the same, at least within the scope of this pull request?
It'd be helpful if we're not introducing breaking API changes from httpx 0.12 -> 0.13 wherever possible. (excepting for example, the UDS support)

Reply:
So this change stems from a [longer conversation with @florimondmanca](https://github.com/encode/httpcore/pull/57#discussion_r407238381). TL;DR is where do we handle the logic behind the `DEFAULT` mode?

I opted for removing it completely as suggested by Florimond and have httpx call into httpcore with a mode explicitly.

I think this way we reduces complexity on httpcore but I don't mind keeping the three modes and the logic behind here if you prefer it. What do you think?